src/test/groovy/net/anshulverma/gradle/estilo/EstiloExtensionTest.groovy

```diff
@@ -43,6 +43,29 @@ class EstiloExtensionTest extends AbstractSpecification {
       checkType        | name
       CheckType.GOOGLE | 'google'
       CheckType.SUN    | 'sun'
+      CheckType.EMPTY  | 'empty'
+      CheckType.CUSTOM | 'custom'
+  }
+
+  def 'Allow custom baseChecks'(){
+    given:
+      def extention = new EstiloExtension()
+      def closure = {
+        customSource checkstylePath
+      }
+      closure.delegate = extention
+
+    when:
+      closure()
+
+    then:
+      extention.baseChecks.filename == checkstylePath
+
+    where:
+      checktype                                 | checkstylePath
+      CheckType.CUSTOM                          | 'config/checkstyle.xml'
```

Comment:
I would have re-used the first test in this file, but I wanted to specifically test that the filename defined in the new field in the extension was added to the CheckType itself.